gms/gossiper.cc

```diff
@@ -643,26 +644,31 @@ future<> gossiper::apply_state_locally(std::map<inet_address, endpoint_state> ma
         std::chrono::steady_clock::now() - start).count());
 }
 
-future<> gossiper::force_remove_endpoint(inet_address endpoint) {
+future<> gossiper::force_remove_endpoint(inet_address endpoint, permit_id pid) {
     if (endpoint == get_broadcast_address()) {
         return make_exception_future<>(std::runtime_error(format("Can not force remove node {} itself", endpoint)));
     }
-    return container().invoke_on(0, [endpoint] (auto& gossiper) mutable -> future<> {
+    return container().invoke_on(0, [endpoint, pid] (auto& gossiper) mutable -> future<> {
+        auto permit = co_await gossiper.lock_endpoint(endpoint, pid);
+        pid = permit.id();
         try {
-            co_await gossiper.remove_endpoint(endpoint);
-            co_await gossiper.evict_from_membership(endpoint);
+            co_await gossiper.remove_endpoint(endpoint, pid);
+            co_await gossiper.evict_from_membership(endpoint, pid);
```

Comment:
Maybe we can do 

co_await gossiper.remove_endpoint(endpoint, permit.id());
co_await gossiper.evict_from_membership(endpoint, permit.id());

to make it less confusing.

Reply:
this is possible.
feel free to send a patch